### ISSUE TITLE
Front Door diagnostic logs not detected when categories are reordered

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#307]
**Description of the issue**

When diagnostic settings are configured to log `FrontdoorAccessLog`. If the category `FrontdoorAccessLog` is returned after `FrontdoorWebApplicationFirewallLog` the rule incorrectly fails.

**Expected behaviour**

The rule should pass if `FrontdoorAccessLog` is configured at all, regardless of order.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.9.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      6.2.4
PSEdition                      Core
GitCommitId                    6.2.4
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
